itests/direct_data_onboard_verified_test.go

```diff
@@ -135,7 +135,7 @@ func TestOnboardRawPieceVerified_WithActorEvents(t *testing.T) {
 	// subscribe to actor events up until the current head
 	initialEventsChan, err := miner.FullNode.SubscribeActorEventsRaw(ctx, &types.ActorEventFilter{
 		FromHeight: epochPtr(0),
-		ToHeight:   epochPtr(int64(head.Height())),
+		ToHeight:   epochPtr(int64(head.Height()) - 1),
```

Comment:
cannot ask for events at tipset >= head because of deferred execution

Reply:
does it error though? this is `Subscribe` and I think it has logic to end the subscription when you reach the limit; at least I remember trying to make that work, so `ToHeight` can be in the future

Reply:
what does this do if you leave it without the `-1`, I think it needs to pass

Reply:
@rvagg It fails during pre-filling as that relies on the historical Index where the filter sanity check forbids a ts >= head. Let me fix it and make this test pass.

Reply:
@rvagg This now passes. 

I have removed the requirement of "maxHeight should be less than head" from the `ChainIndex` Read API as it should be the responsibility of the caller to sanitise the user request anyways.

`eth_GetLogs` already enforces this.

Reply:
nice, so now there's minimal itests modifications